cmd/ctl/pkg/install/install.go

```diff
@@ -0,0 +1,230 @@
+				return nil
+			}
+			options.client.Namespace = settings.Namespace()
+
+			rel, err := options.runInstall(ctx)
+			if err != nil {
+				return err
+			}
+			return writeRelease(ioStreams.Out, rel, options.DryRun)
+		},
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+
+	addInstallUninstallFlags(cmd.Flags(), &options.client.Timeout, &options.client.Wait)
+	addInstallFlags(cmd.Flags(), options.client)
+	addValueOptionsFlags(cmd.Flags(), options.valueOpts)
+	addChartPathOptionsFlags(cmd.Flags(), &options.client.ChartPathOptions)
+
+	cmd.Flags().BoolVar(&options.client.CreateNamespace, "create-namespace", true, "Create the release namespace if not present")
+	cmd.Flags().StringVar(&options.ChartName, "chart-name", "cert-manager", "Name of the chart to install")
+	cmd.Flags().BoolVar(&options.DryRun, "dry-run", false, "Simulate install and output manifest")
+
+	return cmd
+}
+
+func (o *InstallOptions) runInstall(ctx context.Context) (*release.Release, error) {
+	log.SetFlags(0)
+	log.SetOutput(o.Out)
+
+	if err := o.cfg.Init(o.settings.RESTClientGetter(), o.settings.Namespace(), os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
+		return nil, err
+	}
+
+	// Find chart
+	cp, err := o.client.ChartPathOptions.LocateChart(o.ChartName, o.settings)
+	if err != nil {
+		return nil, err
+	}
+
+	chart, err := helm.LoadChart(cp, o.client, o.settings)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if chart is installable
+	if err := helm.CheckIfInstallable(chart); err != nil {
+		return nil, err
+	}
+
+	// Console print if chart is deprecated
+	if chart.Metadata.Deprecated {
+		log.Printf("This chart is deprecated")
+	}
+
+	// Merge all values flags
+	p := getter.All(o.settings)
+	chartValues, err := o.valueOpts.MergeValues(p)
+	if err != nil {
+		return nil, err
+	}
+
+	// Dryrun template generation (used for rendering the crds in /templates)
+	o.client.DryRun = true                  // Do not apply install
+	o.client.IsUpgrade = true               // Do not validate against cluster
+	chartValues[installCRDsFlagName] = true // Make sure to render crds
+	dryRunResult, err := o.client.Run(chart, chartValues)
+	if err != nil {
+		return nil, err
+	}
+
+	if o.DryRun {
+		return dryRunResult, nil
+	}
+
+	// Extract the resource.Info objects from the helm chart crds (/crds folder) and the manifest
+	resources, err := helm.GetChartResourceInfo(dryRunResult.Manifest, o.cfg.KubeClient)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter resource.Info objects and only keep the crds
+	crds := helm.FilterCrdResources(resources)
+
+	// Check if any of these CRDs do already exist
+	installedCrds, err := helm.FetchResources(crds, o.cfg.KubeClient)
+	if err != nil {
+		return nil, err
+	}
+
+	// Abort in case crds are already installed
+	if len(installedCrds) > 0 {
+		return nil, fmt.Errorf("Found existing installed cert-manager crds! Cannot continue with installation.")
+	}
+
+	// Install CRDs
+	if len(crds) > 0 {
```

Comment:
When would this ever be zero? 
The tool should bail out with an error if this is found to be zero, I think.